odoo/models.py

```diff
@@ -6168,8 +6168,11 @@ def filtered(self, func):
             records.filtered("partner_id.is_company")
         """
         if isinstance(func, str):
-            name = func
-            func = lambda rec: any(rec.mapped(name))
+            fnames = func.split('.')
+
+            def func(record):  # pylint: disable-msg=E0102
+                return all((record := record[fname]) for fname in fnames)
+
```

Comment:
> I also wonder if the optimization should not be done by avoiding to call filtered. Rebrowsing si actually kind of expensive at this scale.

You are maybe right, but as far as I remembered the `filtered` was the most costly part of `convert_to_record` and  I focused my effort on `filtered` to improve also more globally the Odoo performance. Anyway, it's can be done in another PR if necessary.

> I also wonder (not sure if it is the case) that we may need to always prefetch active in this case, and assume that it is in cache.

It is already done in the One2Many.read() :wink: (but not for Many2many) 

> It also look like this code (mapped, filtereds, ...) are often modified for micro optimisations
It could be great to add some tests logging timit result, and save statistics on runbot to have a followup of the performance improvement an track that an easier way.

Yep, it is a nice idea.